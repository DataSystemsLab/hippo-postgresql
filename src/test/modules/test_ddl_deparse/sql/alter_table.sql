CREATE TABLE parent (
	a int
);

CREATE TABLE child () INHERITS (parent);

CREATE TABLE grandchild () INHERITS (child);

ALTER TABLE parent ADD COLUMN b serial;

ALTER TABLE parent RENAME COLUMN b TO c;

ALTER TABLE parent ADD CONSTRAINT a_pos CHECK (a > 0);
