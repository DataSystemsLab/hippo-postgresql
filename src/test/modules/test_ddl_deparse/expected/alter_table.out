CREATE TABLE parent (
	a int
);
NOTICE:  DDL test: type simple, tag CREATE TABLE
CREATE TABLE child () INHERITS (parent);
NOTICE:  DDL test: type simple, tag CREATE TABLE
CREATE TABLE grandchild () INHERITS (child);
NOTICE:  DDL test: type simple, tag CREATE TABLE
ALTER TABLE parent ADD COLUMN b serial;
NOTICE:  DDL test: type simple, tag CREATE SEQUENCE
NOTICE:  DDL test: type alter table, tag ALTER TABLE
NOTICE:    subcommand: ADD COLUMN (and recurse)
NOTICE:  DDL test: type simple, tag ALTER SEQUENCE
ALTER TABLE parent RENAME COLUMN b TO c;
NOTICE:  DDL test: type simple, tag ALTER TABLE
ALTER TABLE parent ADD CONSTRAINT a_pos CHECK (a > 0);
NOTICE:  DDL test: type alter table, tag ALTER TABLE
NOTICE:    subcommand: ADD CONSTRAINT (and recurse)
